x/posts/internal/types/post.go

```diff
@@ -90,13 +90,25 @@ func (ids PostIDs) Equals(other PostIDs) bool {
 	return true
 }
 
+// AppendIfMissing appends the given postID to the ids slice if it does not exist inside it yet.
+// It returns a new slice of PostIDs containing such ID and a boolean indicating whether or not the original
+// slice has been modified.
+func (ids PostIDs) AppendIfMissing(id PostID) (PostIDs, bool) {
+	for _, ele := range ids {
+		if ele.Equals(id) {
+			return ids, false
+		}
+	}
+	return append(ids, id), true
+}
+
 // ---------------
 // --- Post
 // ---------------
 
 // Post is a struct of a Magpie post
 type Post struct {
-	PostID            PostID         `json:"id"`                 // Unique id
+	PostID            PostID         `json:"post_id"`            // Unique postID
```

Comment:
Is this necessary? I would prefer we stick to the original one

```suggestion
	PostID            PostID         `json:"id"`            // Unique id
```